airbyte-integrations/connectors/source-trustpilot/source_trustpilot/source.py

```diff
@@ -0,0 +1,178 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+import datetime
+import json
+from dataclasses import InitVar, dataclass, field
+from html.parser import HTMLParser
+from time import sleep
+from typing import Any, Mapping, Optional, Union, Iterable
+from typing import List
+
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.declarative.datetime import MinMaxDatetime
+from airbyte_cdk.sources.declarative.datetime.datetime_parser import DatetimeParser
+from airbyte_cdk.sources.declarative.extractors.record_extractor import RecordExtractor
+from airbyte_cdk.sources.declarative.interpolation import InterpolatedString
+from airbyte_cdk.sources.declarative.requesters.paginators import DefaultPaginator
+from airbyte_cdk.sources.declarative.requesters.request_option import RequestOptionType
+from airbyte_cdk.sources.declarative.stream_slicers import SingleSlice
+from airbyte_cdk.sources.declarative.types import Record, StreamSlice, StreamState, Config
+from airbyte_cdk.sources.declarative.yaml_declarative_source import YamlDeclarativeSource
+
+
+class TrustpilotReviewsParser(HTMLParser):
+    def __init__(self):
+        super().__init__()
+
+        self.__inside_reviews_script = False
+        self.__reviews_script_data = None
+
+    def handle_starttag(self, tag, attrs):
+        if tag == 'script':
+            attrs_map = {e[0]: e[1] for e in attrs}
+            if (('type' in attrs_map) and (attrs_map['type'] == 'application/ld+json')) and \
+                    (('data-business-unit-json-ld' in attrs_map) and (attrs_map['data-business-unit-json-ld'] == 'true')):
+                self.__inside_reviews_script = True
+
+    def handle_data(self, data):
+        if self.__inside_reviews_script:
+            self.__reviews_script_data = data
+
+    def handle_endtag(self, tag):
+        if tag == 'script':
+            if self.__inside_reviews_script:
+                self.__inside_reviews_script = False
+
+    def get_reviews_script_data(self):
+        return self.__reviews_script_data
+
+
+@dataclass
+class TrustpilotExtractor(RecordExtractor):
+    """
+    Extractor for Trustpilot.
+    It parses output HTML and finds all `Review` occurences in it.
+    """
+
+    options: InitVar[Mapping[str, Any]]
+
+    def __post_init__(self, options: Mapping[str, Any]):
+        self._timeout_milliseconds = options['config']['timeout_milliseconds'] if 'timeout_milliseconds' in options['config'] else 1000
+
+    @staticmethod
+    def __parse_html(response: requests.Response) -> List[Record]:
+        dom = response.text
+        parser = TrustpilotReviewsParser()
+        parser.feed(dom)
+        data = parser.get_reviews_script_data()
+        data = json.loads(data)
+        data = data['@graph']
+        data = list(filter(lambda e: e.get('@type', '') == 'Review', data))
+        return data
```

Comment:
@girarda what you think about it? 

@masyagin1998 why are you not using the api endpoints like https://documentation-apidocumentation.trustpilot.com/consumer-api to consume data?

Reply:
@marcosmarxm , @girarda 

It requires API Key, but we can get a lot of data without it

Reply:
IMHO, this implementation must follow default behavior of connectors retrieving data from the provider API and not parsing HTML data. @girarda what your thoughts here?

Reply:
@marcosmarxm I told about it here: https://github.com/airbytehq/connector-contest/issues/122
As I understood RealChrisSean was Okay with it

Reply:
@RealChrisSean @misteryeo This source has clear API endpoints to retrieve data. Of course is easy to implement one single endpoint from the service doing html parsing, but this source has many others which won't be added.
My opinion is this implementation is not following the best practices and how other connectors are doing to retrieve data. Using the API Key is more complex but is the correct way to retrieve data from the service.

Reply:
@girarda this comment here.

Reply:
Reading data from the API is better because we can expect the API to be stable. There are no guarantees around parsing their HTML pages

Reply:
@girarda so is it better now to use it in Airbyte? Sadly, I didn't get credentials for Trustpilot direct account